ethereum/dataproviders/src/main/java/tech/pegasys/teku/dataproviders/lookup/BlockProvider.java

```diff
@@ -37,6 +38,22 @@ static BlockProvider fromDynamicMap(Supplier<Map<Bytes32, SignedBeaconBlock>> ma
     return (roots) -> fromMap(mapSupplier.get()).getBlocks(roots);
   }
 
+  static BlockProvider fromMapWithLock(
```

Comment:
should probably add a test case

Reply:
Thanks you asked for it, I made all stupidly wrong, fixed

Reply:
I did the wrong thing again, thinking how to do it correctly

Reply:
So my understanding on this is:
1. We should remove `SafeFuture.supplyAsync()` to avoid such errors, noone was using it, and we should use our own AsyncRunner whenever it's needed
2. In this case we should use our AsyncRunner 
3. We should also use it in Store methods that use locks and return SafeFuture otherwise they are synchronously locked (`retrieveFinalizedCheckpointAndState`, `createStateGenerationTask`)
4. It will require rework of many `Store` tests
What do you think?

Reply:
i think we could raise a ticket for that cleanup, but probably not worth it now, it'd be a big distraction if its used a bunch in tests...
Using async here though would make sense so we dont add to the problem...